scripts/get_changed_dirs.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+
+#following variable contains the names of the files in the diff with master:
+# i) which are either c, header or preprocesses files in the c folder, and
+# ii) deleted files are not considered
+relevant_diff=`git diff --name-only --diff-filter=d origin/master... -- 'c/*.i' 'c/*.c' 'c/*.h'`
```

Comment:
This script is specific to C tasks, so we should either make it work in general or move it to the `c` directory where we also have other such scripts.

Reply:
I agree with the point. 

I can move the scripts to the c folder. But in this case I would prefer to make a `scripts` (or (`.scripts`) folder in the `c` directory. We already have a couple of scripts there, and with the addition of 1 to 3 more scripts I think it is better to have a directory. We can name it `ci-scripts` (or `.ci-scripts` )to avoid any confusion.

What's your opinion?

Reply:
merging `diff_*.sh` is not that straightforward. `compare.py` does not work in a directory. It always takes all the sets and then goes through all the task definitions and finds out all the files to be checked for preprocessing. At this stage we have added a filter to check if the file is contained in the provided directories or not.

If we go along with the suggestion then either
1. we'll still have to do the first three steps for each directory, or
2.  `compare.py` would need to be changed to preprocess all the files in a given directory

I am not in support of 1, and if we go with 2 then we will still need to address 2 cases of CI: master and branch. I think for the master we still want the script to run as it is, and for the branch we can preprocess all the files in the current directory.

@kfriedberger can comment more.

Reply:
Ok, but changing `compare.py` should be easy. I see no reason not to do this.

Hiding the scripts in `c/.scripts` is not so good, because users are expected to run these checks locally before submitting a PR. Because we still will have only 3 scripts, I think we can keep them in `c/` and refactor later if necessary.

Reply:
OK. Merged the scripts. `compare.py` still needs to be changed.

Reply:
The script `compare.py` has internal dependencies on a hard-coded list of files, like a blacklist of non-comparable tasks or the cfg file containing the architecture for the current task. In comparison to `make` (where each directory has its own set of options in its own makefile), the script `compare.py` needs to be called from the root directory and needs to be used via the set files.